### PR TITLE
PR: Add global and environment configuration paths to load default options from them

### LINKED ISSUE #9805: Update configuration manager to handle site/system level configuration
This needs to define:
- [ ] Locations for system level configuration
- [ ] Locations for defaults (assuming the same folder?)
- [ ] Types of configuration enforcement (what can be changed by the user what cannot)

### LINKED ISSUE #9900: Remove old defaults for spyder
Remove stuff referring to the old configuration system
- [ ] Old defaults
- [ ] Check for config with versions that are relevant for spyder 2 only.

### PR DESCRIPTION
**Pull request by goanpeca**
## Description of Changes

This PR adds support for site configuration files: if a `spyder.ini` file is found on the directories given by `spyder --paths`, those values will be used to replace the default options we provide in Spyder.

Resetting to defaults will reset to these values too.

### Issue(s) Resolved

Fixes #9900 
Fixes #9805

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @goanpeca

### CODE REVIEW

**Comment by goanpeca** [comment]
This should be ready to merge unless you have more comments @ccordoba12

File: spyder/config/manager.py [hunk 1]
```diff
@@ -31,12 +31,30 @@ def __init__(self, parent=None, active_project_callback=None):
...
+                path=site_path,
+                defaults=DEFAULTS,
+                load=False,
+                version=CONF_VERSION,
+                backup=False,
+                raw_mode=True,
+                remove_obsolete=False,
+            )
+            site_defaults = site_config.to_list()
+
```
Comment (goanpeca) [inline spyder/config/manager.py]:
```python
        # Site configuration defines the system defaults if a file
        # is found in the site location
        site_defaults = DEFAULTS
        if os.name == 'nt':
            SEARCH_PATH = (
                'C:/ProgramData/spyder/spyder.ini',
            )
        else:
            SEARCH_PATH = (
                '/etc/spyder/spyder.ini',
                '/var/lib/spyder/spyder.ini',
            )
        site_path = self.get_site_config_path()
        site_fpath = osp.join(site_path, 'spyder.ini')
        SEARCH_PATH += site_fpath

        for fpath in SEARCH_PATH:
            if os.path.isfile(fpath):
                site_config = UserConfig(
                    'spyder',
                    path=fpath,
                    defaults=site_defaults,
                    load=False,
                    version=CONF_VERSION,
                    backup=False,
                    raw_mode=True,
                    remove_obsolete=False,
                )
                site_defaults = site_config.to_list()
```

**Comment by ccordoba12** [comment]
This are the paths returned by `jupyter --paths`:

```
config:
    /home/carlos/.jupyter
    /home/carlos/.virtualenvs/test-ipy3-stable/etc/jupyter
    /usr/local/etc/jupyter
    /etc/jupyter
data:
    /home/carlos/.local/share/jupyter
    /home/carlos/.virtualenvs/test-ipy3-stable/share/jupyter
    /usr/local/share/jupyter
    /usr/share/jupyter
runtime:
    /home/carlos/.local/share/jupyter/runtime
```

**Comment by ccordoba12** [comment]
On Windows this is the output:

```python
config:
    C:\Users\Carlos Cordoba\.jupyter
    C:\Users\Carlos Cordoba\Miniconda2\envs\py36\etc\jupyter
    C:\ProgramData\jupyter
```

File: spyder/config/base.py [hunk 2]
```diff
@@ -228,6 +228,39 @@ def get_conf_path(filename=None):
...
+    CONDA_PREFIX = os.environ.get('CONDA_PREFIX', None)
+
+    if os.name == 'nt':
+        SEARCH_PATH = (
+            'C:/ProgramData/spyder',
+        )
+    else:
+        SEARCH_PATH = (
+            '/etc/spyder',
+            '/usr/local/etc/spyder',
```
Comment (ccordoba12) [inline spyder/config/base.py]: Are these paths the right ones in macOS? What's the output of `jupyter --paths` there?

Comment (goanpeca) [reply spyder/config/base.py]:
```bash
$ jupyter --paths
config:
    /Users/gpena-castellanos/.jupyter
    /Users/gpena-castellanos/anaconda2/envs/spy37/etc/jupyter
    /usr/local/etc/jupyter
    /etc/jupyter
data:
    /Users/gpena-castellanos/Library/Jupyter
    /Users/gpena-castellanos/anaconda2/envs/spy37/share/jupyter
    /usr/local/share/jupyter
    /usr/share/jupyter
runtime:
    /Users/gpena-castellanos/Library/Jupyter/runtime
```

Comment (ccordoba12) [reply spyder/config/base.py]: Thanks for the confirmation @goanpeca!

**Pull request merged**